### DISCUSSION TITLE
Rarity precision

### DISCUSSION THREAD

**Discussion by CryptoPique** [nftchef/art-engine#319] [Help + Q&A]
Hi, I would like to know if there is a way to increase the rarity precision from 0  to 1000 instead of 0 - 100.

could this work ?

const getRarityWeight = (_path) => {
  // check if there is an extension, if not, consider it a directory
  const exp = /#(\d+(?:.\d{1,2})?)/;
  const weight = exp.exec(_path);
  const weightNumber = weight ? parseFloat(weight[1]).toFixed(3) : null;
  if (!weightNumber || isNaN(weightNumber)) {
    return "required";
  }
  return weightNumber;
};

**Comment by nftchef** [comment]
If I understand correctly,

currently, the weight is calculated as `percentage = weight / total sum of weights in a folder`,  it does not need to be out of 100, to use higher numbers, or decimals in the final _chance_ you can supply weights that sum up over 1000 or 10,000 so something like `item#15` in a folder with a  sum of 1,000, would be `15 / 1000 = 1.5%` 

I'm not clear on what you are trying to achieve though 🙇‍♂️

**Comment by CryptoPique** [comment]
ah ok, so rarity is not up to 100 but up to the total sum of # I put
